### PR TITLE
Use up-to-date data when trying to complete shipment

### PR DESCRIPTION
**Pull request by pylipp**
Please see commit messages about details.

The link bug is actually fixed by a one-liner in the first commit. However during debugging I found other places to improve (avoid redundant DB selects and updates, code quality).

### CODE REVIEW

**Comment by pylipp** [comment]
[2.0 BUG: "Remaining Boxes Not Delivered" Action Does Not Successfully Complete Shipment](https://trello.com/c/uAHcqZWf/1321-20-bug-remaining-boxes-not-delivered-action-does-not-successfully-complete-shipment)

File: back/boxtribute_server/dev_main.py [hunk 1]
```diff
@@ -3,3 +3,7 @@
 from .routes import api_bp, app_bp
 
 app = main(api_bp, app_bp)
+
+
+if __name__ == "__main__":
+    app.run("0.0.0", 5005, debug=True)
```
[the remark this answers is not in the document]
Comment (HaGuesto) [reply back/boxtribute_server/dev_main.py:9]: @pylipp

**Review by HaGuesto — APPROVED** [review]
I guess you went down a rabbit hole, @pylipp?

Feel free to merge after removing the debug state

File: back/boxtribute_server/dev_main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pylipp) [reply back/boxtribute_server/dev_main.py]: @HaGuesto this can be ignored because `dev_main.py` is used for development only.

**Pull request merged**